### ISSUE TITLE
inference speed on an edgetpu

### ISSUE THREAD

**Issue by kaishijeng** [markus-k/ultrafast-lane-detection-tf2#1]
@markus-k 

Can it run realtime (30fps) o an edgetpu?
Can you share a pretrained model which I can try it?

Thanks,

**Comment by markus-k** [comment]
Hi,

from some first tests, we are getting about 20 FPS on single EdgeTPU. Some optimizations can probably still be done. I should point out that the code in this repo is nowhere near complete, only the Model itself should be implemented correctly so far. We are still missing correctly implemented training-loss and data generation, I hope we can get something done in the upcoming weeks.

Our goal is to use this on an embedded platform in real time, so we are obviously in the hunt for performance as well. 

Here is a pretrained model using the LLAMAS dataset for running on the EdgeTPU: [ultrafast_quant_edgetpu.tflite.zip](https://github.com/markus-k/ultrafast-lane-detection-tf2/files/4962119/ultrafast_quant_edgetpu.tflite.zip). It is working fairly good on the LLAMAS test data, but be aware that this model doesn't quite match the paper yet, and is probably terribly overfitted.

**Comment by kaishijeng** [comment]
I will measure fps on my edgetpu system (PC with Coral USB dongle)  with your pretrained model.
Is your edgetpu system  RPI or PC?

Thanks,

**Comment by markus-k** [comment]
Currently on PC (Mac), but it should run on an ARM system later.

**Comment by busyyang** [comment]
Hi, I test the _edgetpu.tflite model as you mentioned above. But the output seems not good. 
Input:
![2](https://user-images.githubusercontent.com/7272659/129831379-a7c8ae92-4166-4c9e-b244-ada34472eca1.jpg)

Output:
![微信截图_20210818111800](https://user-images.githubusercontent.com/7272659/129831432-eff0fe91-10a4-434b-b705-3717994c4cab.png)

The test code is as below:
~~~py
# encoding: utf-8
import tflite_runtime.interpreter as tflite
import platform, cv2
import numpy as np

EDGETPU_SHARED_LIB = {
    'Linux': 'libedgetpu.so.1',
    'Darwin': 'libedgetpu.1.dylib',
    'Windows': 'edgetpu.dll'
}[platform.system()]


class Net():
    def __init__(self, lite_model, use_tpu):
        if use_tpu:
            try:
                interpreter = tflite.Interpreter(model_path=lite_model,
                                                 experimental_delegates=[tflite.load_delegate(EDGETPU_SHARED_LIB)])
                print('Load model {} to TPU device'.format(lite_model))
            except Exception as e:
                print("Can't load `{}` model to tpu. Please try to run without tpu.".format(lite_model))
                exit(-2)
        else:
            interpreter = tflite.Interpreter(model_path=lite_model)
        interpreter.allocate_tensors()
        input_details = interpreter.get_input_details()
        output_details = interpreter.get_output_details()

        self.interpreter = interpreter
        self.input_details = input_details
        self.output_details = output_details

    def predict(self, input):
        if len(self.input_details[0]['shape']) - len(input.shape) == 1:
            input = np.expand_dims(input, axis=0)
        self.interpreter.set_tensor(self.input_details[0]['index'], input.astype('float32'))
        self.interpreter.invoke()
        output = self.interpreter.get_tensor(self.output_details[0]['index'])
        return output


model = Net('./models/ultrafast_quant_edgetpu.tflite', True)

input_image = cv2.imread('imgs/2.jpg')
h, w, _ = input_image.shape
input_image = cv2.cvtColor(input_image, cv2.COLOR_BGR2RGB)
input_image = cv2.resize(input_image, (800, 288)) / 255.

y = model.predict(input_image)
y = y[0]
max_val = np.max(y)
min_val = np.min(y)
y = (y - min_val) / (max_val - min_val) * 255
y = y.astype('uint8')
y = cv2.resize(y, (w, h))
cv2.imshow('sss', y[..., 0])
cv2.waitKey(0)

~~~




And, the values of output is seems like 1.6678208 or times of it. what is the real meaning of these values? can you share your trained `.h5` or cpu `.tflite` model?
